src/test/java/io/jenkins/plugins/folderauth/FolderAuthorizationWebAPITest.java

```diff
@@ -0,0 +1,306 @@
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) jenkinsRule.jenkins.getAuthorizationStrategy();
+        for (AgentRole agentRole : strategy.getAgentRoles()) {
+            assertNotEquals("globalRole", agentRole.getName());
+        }
+    }
+
+    private enum RoleType {
+        GLOBAL, FOLDER, AGENT
+    }
+
+    /**
+     * Util method to add a role and verify that the request succeeded
+     * @param roleType Type of role (either GLOBAL, FOLDER or AGENT
+     * @param jsonString String containing the json representation of a role
+     * @throws IOException
+     */
+    private void addRole(RoleType roleType, String jsonString) throws IOException {
+        String apiPath;
+        switch (roleType) {
+            case GLOBAL:
+                apiPath = "folder-auth/addGlobalRole";
+                break;
+            case FOLDER:
+                apiPath = "folder-auth/addFolderRole";
+                break;
+            case AGENT:
+                apiPath = "folder-auth/addAgentRole";
+                break;
+            default:
+                throw new IllegalArgumentException("Unexpected role type");
+        }
+        URL apiURL = new URL(jenkinsRule.jenkins.getRootUrl() + apiPath);
+        WebRequest request = new WebRequest(apiURL, HttpMethod.POST);
+        request.setRequestBody(jsonString);
+        request.setAdditionalHeader("Content-Type", "application/json");
+        Page page = webClient.getPage(request);
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+    }
+
+    /**
+     * Util method to verify that the role exists
+     * @param roleType Type of role
+     * @param roleMap Map object that represents the role
+     * @throws NoSuchMethodException
+     * @throws InvocationTargetException
+     * @throws IllegalAccessException
+     */
+    private void verifyThatRoleExists(RoleType roleType, Map<String, Object> roleMap)
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        AuthorizationStrategy a = jenkinsRule.jenkins.getAuthorizationStrategy();
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) a;
+        boolean found = false;
+        // Using reflection to get method
+        String getRoleMethodString = "get" + StringUtils.capitalize(roleType.name().toLowerCase()  + "Roles");
+        Method getRoleMethod = strategy.getClass().getMethod(getRoleMethodString);
+        Set<AbstractRole> roleSet = (Set) getRoleMethod.invoke(strategy);
+        for (AbstractRole role : roleSet) {
+            if (role.getName().equals(roleMap.get("name"))) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
```

Comment:
I think that this function and the `verifyUserAssignedToRole` could just return the boolean value. We can then also check that the role was not present before the API call.